### PR TITLE
Migrate WFE to the new bootstrap mechanism

### PR DESCRIPTION
**Pull request by ChristophWurst**

### CODE REVIEW

File: apps/workflowengine/lib/Listener/LoadAdditionalSettingsScriptsListener.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+use function script;
+
+class LoadAdditionalSettingsScriptsListener implements IEventListener {
+	public function handle(Event $event): void {
+		if (!function_exists('style')) {
+			// This is hacky, but we need to load the template class
+			class_exists(Template::class, true);
+		}
+
+		script('core', [
```
Comment (kesselb) [inline apps/workflowengine/lib/Listener/LoadAdditionalSettingsScriptsListener.php:43]: `OC_Util::addStyle` might work without the class exists hack ;)

Comment (blizzz) [reply apps/workflowengine/lib/Listener/LoadAdditionalSettingsScriptsListener.php:43]: which?

**Review by blizzz — CHANGES_REQUESTED** [review]
```json
{
  "reqId": "Fxy8iwmGMpA3GqFnVLBs",
  "level": 3,
  "time": "2020-06-23T21:17:44+00:00",
  "remoteAddr": "127.0.0.1",
  "user": "master",
  "app": "no app in context",
  "method": "GET",
  "url": "/master/index.php/settings/user/workflow",
  "message": {
    "Exception": "OCP\\AppFramework\\QueryException",
    "Message": "Could not resolve OCA\\Workflowengine\\Listener\\LoadAdditionalSettingsScriptsListener! Class OCA\\Workflowengine\\Listener\\LoadAdditionalSettingsScriptsListener does not exist",
    "Code": 0,
    "Trace": [
      {
        "file": "/srv/http/nextcloud/master/lib/private/AppFramework/Utility/SimpleContainer.php",
        "line": 116,
        "function": "resolve",
        "class": "OC\\AppFramework\\Utility\\SimpleContainer",
        "type": "->",
        "args": [
          "OCA\\Workflowengine\\Listener\\LoadAdditionalSettingsScriptsListener"
        ]
      },
      {
        "file": "/srv/http/nextcloud/master/lib/private/ServerContainer.php",
        "line": 149,
        "function": "query",
        "class": "OC\\AppFramework\\Utility\\SimpleContainer",
        "type": "->",
        "args": [
          "OCA\\Workflowengine\\Listener\\LoadAdditionalSettingsScriptsListener",
          true
        ]
      },
      {
        "file": "/srv/http/nextcloud/master/lib/private/EventDispatcher/ServiceEventListener.php",
        "line": 66,
        "function": "query",
        "class": "OC\\ServerContainer",
        "type": "->",
        "args": [
          "OCA\\Workflowengine\\Listener\\LoadAdditionalSettingsScriptsListener"
        ]
      },
      {
        "file": "/srv/http/nextcloud/master/3rdparty/symfony/event-dispatcher/EventDispatcher.php",
        "line": 251,
        "function": "__invoke",
        "class": "OC\\EventDispatcher\\ServiceEventListener",
        "type": "->",
        "args": [
          {
            "__class__": "OCP\\WorkflowEngine\\Events\\LoadSettingsScriptsEvent"
          },
          "OCP\\WorkflowEngine::loadAdditionalSettingScripts",
          {
            "__class__": "Symfony\\Component\\EventDispatcher\\EventDispatcher"
          }
        ]
      },
      {
        "file": "/srv/http/nextcloud/master/3rdparty/symfony/event-dispatcher/EventDispatcher.php",
        "line": 73,
        "function": "callListeners",
        "class": "Symfony\\Component\\EventDispatcher\\EventDispatcher",
        "type": "->",
        "args": [
          [
            {
              "__class__": "Closure"
            }
          ],
          "OCP\\WorkflowEngine::loadAdditionalSettingScripts",
          {
            "__class__": "OCP\\WorkflowEngine\\Events\\LoadSettingsScriptsEvent"
          }
        ]
      },
      {
        "file": "/srv/http/nextcloud/master/lib/private/EventDispatcher/EventDispatcher.php",
        "line": 86,
        "function": "dispatch",
        "class": "Symfony\\Component\\EventDispatcher\\EventDispatcher",
        "type": "->",
        "args": [
          {
            "__class__": "OCP\\WorkflowEngine\\Events\\LoadSettingsScriptsEvent"
          },
          "OCP\\WorkflowEngine::loadAdditionalSettingScripts"
        ]
      },
      {
        "file": "/srv/http/nextcloud/master/apps/workflowengine/lib/Settings/ASettings.php",
        "line": 87,
        "function": "dispatch",
        "class": "OC\\EventDispatcher\\EventDispatcher",
        "type": "->",
        "args": [
          "OCP\\WorkflowEngine::loadAdditionalSettingScripts",
          {
            "__class__": "OCP\\WorkflowEngine\\Events\\LoadSettingsScriptsEvent"
          }
        ]
      },
      {
        "file": "/srv/http/nextcloud/master/apps/settings/lib/Controller/CommonSettingsTrait.php",
        "line": 141,
        "function": "getForm",
        "class": "OCA\\WorkflowEngine\\Settings\\ASettings",
        "type": "->",
        "args": []
      },
      {
        "file": "/srv/http/nextcloud/master/apps/settings/lib/Controller/PersonalSettingsController.php",
        "line": 76,
        "function": "formatSettings",
        "class": "OCA\\Settings\\Controller\\PersonalSettingsController",
        "type": "->",
        "args": [
          [
            [
              {
                "__class__": "OCA\\WorkflowEngine\\Settings\\Personal"
              }
            ]
          ]
        ]
      },
      {
        "file": "/srv/http/nextcloud/master/apps/settings/lib/Controller/CommonSettingsTrait.php",
        "line": 152,
        "function": "getSettings",
        "class": "OCA\\Settings\\Controller\\PersonalSettingsController",
        "type": "->",
        "args": [
          "workflow"
        ]
      },
      {
        "file": "/srv/http/nextcloud/master/apps/settings/lib/Controller/PersonalSettingsController.php",
        "line": 67,
        "function": "getIndexResponse",
        "class": "OCA\\Settings\\Controller\\PersonalSettingsController",
        "type": "->",
        "args": [
          "personal",
          "workflow"
        ]
      },
      {
        "file": "/srv/http/nextcloud/master/lib/private/AppFramework/Http/Dispatcher.php",
        "line": 169,
        "function": "index",
        "class": "OCA\\Settings\\Controller\\PersonalSettingsController",
        "type": "->",
        "args": [
          "workflow"
        ]
      },
      {
        "file": "/srv/http/nextcloud/master/lib/private/AppFramework/Http/Dispatcher.php",
        "line": 100,
        "function": "executeController",
        "class": "OC\\AppFramework\\Http\\Dispatcher",
        "type": "->",
        "args": [
          {
            "__class__": "OCA\\Settings\\Controller\\PersonalSettingsController"
          },
          "index"
        ]
      },
      {
        "file": "/srv/http/nextcloud/master/lib/private/AppFramework/App.php",
        "line": 137,
        "function": "dispatch",
        "class": "OC\\AppFramework\\Http\\Dispatcher",
        "type": "->",
        "args": [
          {
            "__class__": "OCA\\Settings\\Controller\\PersonalSettingsController"
          },
          "index"
        ]
      },
      {
        "file": "/srv/http/nextcloud/master/lib/private/AppFramework/Routing/RouteActionHandler.php",
        "line": 47,
        "function": "main",
        "class": "OC\\AppFramework\\App",
        "type": "::",
        "args": [
          "OCA\\Settings\\Controller\\PersonalSettingsController",
          "index",
          {
            "__class__": "OC\\AppFramework\\DependencyInjection\\DIContainer"
          },
          {
            "section": "workflow",
            "_route": "settings.PersonalSettings.index"
          }
        ]
      },
      {
        "function": "__invoke",
        "class": "OC\\AppFramework\\Routing\\RouteActionHandler",
        "type": "->",
        "args": [
          {
            "section": "workflow",
            "_route": "settings.PersonalSettings.index"
          }
        ]
      },
      {
        "file": "/srv/http/nextcloud/master/lib/private/Route/Router.php",
        "line": 297,
        "function": "call_user_func",
        "args": [
          {
            "__class__": "OC\\AppFramework\\Routing\\RouteActionHandler"
          },
          {
            "section": "workflow",
            "_route": "settings.PersonalSettings.index"
          }
        ]
      },
      {
        "file": "/srv/http/nextcloud/master/lib/base.php",
        "line": 1010,
        "function": "match",
        "class": "OC\\Route\\Router",
        "type": "->",
        "args": [
          "/settings/user/workflow"
        ]
      },
      {
        "file": "/srv/http/nextcloud/master/index.php",
        "line": 37,
        "function": "handleRequest",
        "class": "OC",
        "type": "::",
        "args": []
      }
    ],
    "File": "/srv/http/nextcloud/master/lib/private/AppFramework/Utility/SimpleContainer.php",
    "Line": 107,
    "CustomMessage": "Could not load event listener service OCA\\Workflowengine\\Listener\\LoadAdditionalSettingsScriptsListener"
  },
  "userAgent": "Mozilla/5.0 (X11; Linux x86_64; rv:77.0) Gecko/20100101 Firefox/77.0",
  "version": "20.0.0.0"
}
```

**Comment by ChristophWurst** [comment]
I forgot the autoloader update!

**Comment by blizzz** [comment]
> I forgot the autoloader update!

i thought so. Weirldy enough, when i ran the checker locally last night, it didn't add the files :shrug:

**Comment by ChristophWurst** [comment]
> i thought so. Weirldy enough, when i ran the checker locally last night, it didn't add the files

The namespace had incorrect casing.

I've rebased & squashed, including your changes :)

File: apps/workflowengine/lib/Listener/LoadAdditionalSettingsScriptsListener.php (hunk 1 above)
Comment (ChristophWurst) [reply apps/workflowengine/lib/Listener/LoadAdditionalSettingsScriptsListener.php:43]: Yeah, though unrelated to this change. Let's do it in a separate PR :)

Comment (kesselb) [reply apps/workflowengine/lib/Listener/LoadAdditionalSettingsScriptsListener.php:43]: Fine by me ;)

**Review by blizzz — APPROVED** [review]

**Review by rullzer — APPROVED** [review]

**Pull request merged**